src/utils/file_utils.py

```diff
@@ -59,26 +65,62 @@ def rename_json_if_exists(filepath: str, new_filepath: str):
         print(f"No corresponding JSON file found for {filepath}")
 
 
-def exclude_files(files, platforms, skip_queued, skip_posted=True):
+def get_excluded_tags(account: Account, skip_posted: bool, skip_queued: bool):
+    # Get tags which will be excluded from the results
     excluded_tags = []
     if skip_posted:
-        excluded_tags = [POSTED_TAG_MAPPING[platform] for platform in platforms]
+        excluded_tags.extend([POSTED_TAG_MAPPING[platform] for platform in account.platforms])
     if skip_queued:
-        excluded_tags.extend([QUEUE_TAG_MAPPING[platform] for platform in platforms])
-    return [f for f in files if not any(tag in f for tag in excluded_tags)]
+        excluded_tags.extend([QUEUE_TAG_MAPPING[platform] for platform in account.platforms])
+    return excluded_tags
+
+
+def matches_path(file, pattern):
+    return fnmatch.fnmatch(file, f"{pattern}/*")
+
+
+def excluded_via_scheduler_profile_paths(account: Account, file: str):
+    # If none of the specified schedule profiles has a path that matches
+    # the file, exclude it
+    if len(account.scheduler_profiles) == 0:
+        return False
+    return not any(matches_path(file, scheduler_profile.directory_path) for scheduler_profile in account.scheduler_profiles)
+
+
+def excluded_via_scheduler_profile_exclusions(account: Account, file: str):
+    # If the path is excluded explicitly via a schedule profile, exclude it
+    if len(account.scheduler_profiles) == 0:
+        return False
+
+    for scheduler_profile in account.scheduler_profiles:
+        if any(matches_path(file, exclude_path) for exclude_path in scheduler_profile.exclude_paths):
+            return True
+    return False
+
+
+def is_excluded_file(account: Account, file: str, excluded_tags: List[str]):
```

Comment:
idea: turn each dir_path and excluded_dir path into 2 arrays, match on those
eg
pattern with excluded_tag # 1 
pattern with excluded tag # 2
excluded schedule profile # 1
 etc